packages/build/src/run-draft.spec.ts

```diff
@@ -0,0 +1,190 @@
+      expect(uploadArtifactToDownloadCenter).to.have.been.callCount(ALL_BUILD_VARIANTS.length);
+      expect(uploadReleaseAsset).to.have.been.callCount(ALL_BUILD_VARIANTS.length);
+    });
+
+    it('does not run when there is no triggering tag', async() => {
+      const uploadReleaseAsset = sinon.stub();
+      githubRepo = createStubRepo({
+        uploadReleaseAsset
+      });
+
+      await runDraft(
+        config,
+        githubRepo,
+        uploadArtifactToDownloadCenter,
+        downloadArtifactFromEvergreen,
+        ensureGithubReleaseExistsAndUpdateChangelog
+      );
+      expect(ensureGithubReleaseExistsAndUpdateChangelog).to.not.have.been.called;
+      expect(downloadArtifactFromEvergreen).to.not.have.been.called;
+      expect(uploadArtifactToDownloadCenter).to.not.have.been.called;
+      expect(uploadReleaseAsset).to.not.have.been.called;
+    });
+
+    it('fails if packageInformation is missing', async() => {
+      const uploadReleaseAsset = sinon.stub();
+      githubRepo = createStubRepo({
+        uploadReleaseAsset
+      });
+      config.triggeringGitTag = 'v0.8.1-draft.2';
+      config.packageInformation = undefined;
+
+      try {
+        await runDraft(
+          config,
+          githubRepo,
+          uploadArtifactToDownloadCenter,
+          downloadArtifactFromEvergreen,
+          ensureGithubReleaseExistsAndUpdateChangelog
+        );
+      } catch (e) {
+        expect(e.message).to.contain('Missing package information from config');
+        expect(ensureGithubReleaseExistsAndUpdateChangelog).to.not.have.been.called;
+        expect(downloadArtifactFromEvergreen).to.not.have.been.called;
+        expect(uploadArtifactToDownloadCenter).to.not.have.been.called;
+        expect(uploadReleaseAsset).to.not.have.been.called;
+        return;
+      }
+      expect.fail('Expected error');
+    });
+  });
+
+  describe('ensureGithubReleaseExistsAndUpdateChangelog', () => {
+    let generateChangelog: typeof generateChangelogFn;
+
+    beforeEach(() => {
+      generateChangelog = sinon.stub().returns('Nice changelog');
+    });
+
+    it('generates the release notes and updates the release', async() => {
+      const getPreviousReleaseTag = sinon.stub().resolves('v0.8.0');
+      const updateDraftRelease = sinon.stub().resolves();
+      githubRepo = createStubRepo({
+        getPreviousReleaseTag, updateDraftRelease
+      });
+
+      await ensureGithubReleaseExistsAndUpdateChangelogFn(
+        '0.8.1', 'v0.8.1', githubRepo, generateChangelog
+      );
+      expect(getPreviousReleaseTag).to.have.been.called;
+      expect(generateChangelog).to.have.been.called;
+      expect(updateDraftRelease).to.have.been.calledWith({
+        name: '0.8.1',
+        tag: 'v0.8.1',
+        notes: 'Release notes [in Jira](https://jira.mongodb.org/issues/?jql=project%20%3D%20MONGOSH%20AND%20fixVersion%20%3D%200.8.1)\n\nNice changelog'
```

Comment:
shouldn't this be 'Nice changelog'?

Reply:
No, only the `changelog` function in the `git` package generates the pure commit stuff. The `ensure...` also adds the release notes link (it didn't make sense for me to include this part into `git`, as it is not git relevant).

Reply:
What I meant is, do we keep the `Release notes [in Jira](https://jira.mongodb.org/issues/?jql=project%20%3D%20MONGOSH%20AND%20fixVersion%20%3D%200.8.1)` too?

Maybe we could move them something like in the end / make sure is not confusing what's what? I guess it doesn't make much of a difference though.

Reply:
Ah understood... I rephrased it to: `See an overview of all solved issues in Jira` and append it instead of prepend it to the changelog.